presto-main/src/main/java/com/facebook/presto/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -82,6 +82,8 @@ public static Expression extract(PlanNode node)
                 return new ComparisonExpression(ComparisonExpressionType.EQUAL, reference, expression);
             };
 
+    private EffectivePredicateExtractor() {}
```

Comment:
the best would be to have `PlanVisitor<Expression, Void>` be implemented as inner static class

Reply:
I didn't want to change the whole class. But yet, this can be done.